### PR TITLE
Feat: Handle maintenance in `UptimeCalculator`

### PR DESCRIPTION
**Pull request by chakflying**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

Save maintenance heartbeats in an extras column of the stats tables.

## Type of change

- New feature (non-breaking change which adds functionality)
- Breaking change (a fix or feature that would cause existing functionality to not work as expected)
- Other

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [ ] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

### CODE REVIEW

**Comment by chakflying** [comment]
Also found an edge case bug: If a push monitor is under maintenance, every push call would trigger a toast notification on the client.

File: server/uptime-calculator.js [hunk 1]
```diff
@@ -606,7 +654,11 @@ class UptimeCalculator {
             avgPing = totalPing / total.up;
         }
 
-        uptimeData.uptime = total.up / (total.up + total.down);
+        if (total.up + total.down === 0) {
+            uptimeData.uptime = 0;
```
Comment (CommanderStorm) [inline server/uptime-calculator.js:658]: I think setting it to `undefined`/`null` and handling this in the frontend might be a better approach, but this is a good enough interims solution ^^

**Review by CommanderStorm — COMMENTED** [review]
overall, these changes are very reasonable.
I have left a few stylistic changes and a question about the closures

File: server/uptime-calculator.js [hunk 2]
```diff
@@ -263,6 +290,13 @@ class UptimeCalculator {
         dailyStatBean.ping = dailyData.avgPing;
         dailyStatBean.pingMin = dailyData.minPing;
         dailyStatBean.pingMax = dailyData.maxPing;
+        {
```
[the remark this answers is not in the document]
Comment (chakflying) [reply server/uptime-calculator.js:293]: The properties in the object have the same name as some other variables in the outer function, causing a conflict. I'm not sure if you can change the variable name while using destructuring assignment, but enclosing this in a block was the easiest solution I could think of.

**Comment by CommanderStorm** [comment]
Have tested the fix and it works.
What do you think @chakflying?

**Pull request merged**